p2p/starknet/starknet_test.go

```diff
@@ -97,14 +98,42 @@ func TestClientHandler(t *testing.T) {
 	})
 
 	t.Run("get events", func(t *testing.T) {
-		res, cErr := client.RequestEvents(testCtx, &spec.EventsRequest{})
+		const limit uint64 = 3
+		const firstBlock uint64 = 0
+
+		for blockNumber := firstBlock; blockNumber < limit; blockNumber++ {
+			mockReader.EXPECT().BlockByNumber(blockNumber).Return(&core.Block{
+				Header: &core.Header{
+					Number: blockNumber,
+				},
+				Receipts: []*core.TransactionReceipt{
+					{Events: []*core.Event{}},
```

Comment:
Let's add some events here and make sure they are adapted properly? We will cover AdaptEvents as well that way.
Like, start from no events and add increasingly more events each block.